nucypher/blockchain/eth/interfaces.py

```diff
@@ -597,3 +513,58 @@ def deploy_contract(self,
                                  contract_abi=contract_factory.abi)
 
         return contract, txhash  # receipt
+
+    def get_contract_factory(self, contract_name: str) -> Contract:
+        """Retrieve compiled interface data from the cache and return web3 contract"""
+        try:
+            interface = self.__raw_contract_cache[contract_name]
+        except KeyError:
+            raise self.UnknownContract('{} is not a locally compiled contract.'.format(contract_name))
+        except TypeError:
+            if self.__raw_contract_cache is NO_COMPILATION_PERFORMED:
+                message = "The local contract compiler cache is empty because no compilation was performed."
+                raise self.InterfaceError(message)
+            raise
+        else:
+            contract = self.client.w3.eth.contract(abi=interface['abi'],
+                                                   bytecode=interface['bin'],
+                                                   ContractFactoryClass=Contract)
+            return contract
+
+    def _wrap_contract(self, wrapper_contract: Contract, target_contract: Contract) -> Contract:
+        """
+        Used for upgradeable contracts; Returns a new contract object assembled
+        with its own address but the abi of the other.
+        """
+
+        # Wrap the contract
+        wrapped_contract = self.client.w3.eth.contract(abi=target_contract.abi,
+                                                       address=wrapper_contract.address,
+                                                       ContractFactoryClass=self._contract_factory)
+        return wrapped_contract
+
+    def get_proxy(self, target_address: str, proxy_name: str) -> Contract:
+
+        # Lookup proxies; Search for a registered proxy that targets this contract record
+        records = self.registry.search(contract_name=proxy_name)
+
+        dispatchers = list()
+        for name, addr, abi in records:
+            proxy_contract = self.client.w3.eth.contract(abi=abi,
```

Comment:
```suggestion
            proxy_contract = self.client.get_contract(abi=abi,
```

Reply:
You know, this one I'm tempted to leave as direct access to `w3`, since it's so low-level within out API.

Reply:
I guess it is within the `BlockchainInterface`